### PR TITLE
Move database settings from models to setting

### PR DESCRIPTION
**Pull request by lunny**
And also added `MAX_IDLE_CONNS` and `CONN_MAX_LIFETIME` options.

### CODE REVIEW

File: models/models_test.go [hunk 1]
```diff
@@ -11,99 +10,19 @@ import (
...
-}
-
 func TestDumpDatabase(t *testing.T) {
 	assert.NoError(t, PrepareTestDatabase())
 
 	dir, err := ioutil.TempDir(os.TempDir(), "dump")
 	assert.NoError(t, err)
 
-	for _, dbType := range supportedDatabases {
+	for _, dbType := range setting.SupportedDatabases {
```
Comment (sapk) [inline models/models_test.go]: Small nit but ease the understanding of code
```suggestion
	for _, dbName := range setting.SupportedDatabases {
		dbType := setting.GetDBTypeByName(dbName)
```

File: modules/setting/database.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+var (
+	// SupportedDatabases includes all supported databases type
+	SupportedDatabases = []string{"MySQL", "PostgreSQL", "MSSQL"}
+	dbTypes            = map[string]string{"MySQL": "mysql", "PostgreSQL": "postgres", "MSSQL": "mssql", "SQLite3": "sqlite3"}
+
+	// EnableSQLite3 use SQLite3, set by build flag
+	EnableSQLite3 bool
+
+	// EnableTiDB enable TiDB, set by build flag
+	EnableTiDB bool
```
Comment (techknowlogick) [inline modules/setting/database.go]: no longer supported

File: modules/setting/database.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+		connStr = fmt.Sprintf("server=%s; port=%s; database=%s; user id=%s; password=%s;", host, port, Database.Name, Database.User, Database.Passwd)
+	case "sqlite3":
+		if !EnableSQLite3 {
+			return "", errors.New("this binary version does not build support for SQLite3")
+		}
+		if err := os.MkdirAll(path.Dir(Database.Path), os.ModePerm); err != nil {
+			return "", fmt.Errorf("Failed to create directories: %v", err)
+		}
+		connStr = fmt.Sprintf("file:%s?cache=shared&mode=rwc&_busy_timeout=%d", Database.Path, Database.Timeout)
+	case "tidb":
```
Comment (techknowlogick) [inline modules/setting/database.go]: no longer supported

File: routers/init.go [hunk 4]
```diff
@@ -102,10 +102,10 @@ func GlobalInit() {
 		models.InitDeliverHooks()
 		models.InitTestPullRequests()
 	}
-	if models.EnableSQLite3 {
+	if setting.EnableSQLite3 {
 		log.Info("SQLite3 Supported")
 	}
-	if models.EnableTiDB {
+	if setting.EnableTiDB {
```
Comment (techknowlogick) [inline routers/init.go]: no longer supported

**Comment by lunny** [comment]
@aspk @techknowlogick  done.

**Review by techknowlogick — APPROVED** [review]

**Review by sapk — APPROVED** [review]

**Pull request merged**